gadgets/ci/image_inspect/test/unit/image_inspect_test.go

```diff
@@ -0,0 +1,203 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+	"sort"
+	"testing"
+
+	gadgettesting "github.com/inspektor-gadget/inspektor-gadget/gadgets/testing"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/environment/local"
+	gadgetcontext "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context"
+	apihelpers "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api-helpers"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/traceloop/tracer"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	ocihandler "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/oci-handler"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/local"
+
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/ebpf"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/wasm"
+)
+
+type extraInfoMap struct {
+	Name string
+	Type string
+}
+
+type extraInfoProgram struct {
+	Section string
+	Source  string
+}
+
+type extraInfoVariable struct {
+	Name   string
+	Offset uint64
+	Size   uint64
+	Map    string
+}
+
+func TestInspectCmd(t *testing.T) {
+	gadgettesting.InitUnitTest(t)
+	t.Parallel()
+
+	image := "image_inspect"
+	if _, ok := os.LookupEnv("GADGET_REPOSITORY"); !ok {
+		image = fmt.Sprintf("ghcr.io/inspektor-gadget/gadget/%s", image)
+	}
+
+	opGlobalParams := make(map[string]*params.Params)
+	ociParams := apihelpers.ToParamDescs(ocihandler.OciHandler.InstanceParams()).ToParams()
+
+	ocihandler.OciHandler.Init(ociParams)
+
+	opGlobalParams["oci"] = apihelpers.ToParamDescs(ocihandler.OciHandler.GlobalParams()).ToParams()
+	opGlobalParams["oci"].Set("verify-image", "false")
+
+	runtime := local.New()
+	runtimeParams := runtime.ParamDescs().ToParams()
+
+	ops := make([]operators.DataOperator, 0)
+	err := ocihandler.OciHandler.Init(opGlobalParams["oci"])
+	if err != nil {
+		t.Fatalf("Error initializing OCI handler: %v\n", err)
+		return
+	}
+	ops = append(ops, ocihandler.OciHandler)
+
+	gadgetCtx := gadgetcontext.New(
+		context.Background(),
+		image,
+		gadgetcontext.WithDataOperators(ops...),
+		gadgetcontext.WithUseInstance(false),
+		gadgetcontext.IncludeExtraInfo(true),
+	)
+
+	paramValueMap := make(map[string]string)
+	ociParams.CopyToMap(paramValueMap, "operator.oci.")
+
+	info, err := runtime.GetGadgetInfo(gadgetCtx, runtimeParams, paramValueMap)
```

Comment:
Please add a very first test that validate that `info.ExtraInfo.Data` contains all the keys that we expect.